pipeline_dp/aggregate_params.py

```diff
@@ -650,18 +650,44 @@ class AddDPNoiseParams:
     sensitivity by contribution bounding and relies on the caller to ensure the
     provided data satisfies the provided bound.
 
-   Attributes:
+    The noise in DP is scaled with sensitivity, where sensitivity for query q is
+     sensitivity = \max_{D, D' neighbouring} ||q(D)-q(D')||
+    Where ||.|| should be a proper norm, L1 for Laplace Mechanism, L2 for
+    Gaussian Mechanism.
+    See https://en.wikipedia.org/wiki/Norm_(mathematics)#p-norm on norm details.
+
+    In our case, q values are represented as collection of
+    (partition_key, value). Where partition_key is any identifier, value is
+    a scalar. q values can be represented as a vector indexed by partition_key.
+    One of the convenient ways to define l1/l2 sensitivity is through l0/linf
+    sensitivities.
+    Namely
+
+    l1_sensitivity = l0_sensitivity*linf_sensitivity
+    l2_sensitivity = sqrt(l0_sensitivity)*linf_sensitivity
+
+    For applying the Laplace mechanism or l1_sensitivity or
+    l0_sensitivity, linf_sensitivity must be specified.
+    
+    For applying the Gaussian mechanism or l2_sensitivity or
+    l0_sensitivity, linf_sensitivity must be specified.
+
+    Attributes:
        noise_kind: The type of noise to use for the DP calculations.
        l0_sensitivity: the maximum number of partition for which 1 privacy unit
          can contribute.
        linf_sensitivity: the maximum difference of values in one partition which
          can achieved by adding or removing one privacy unit from the dataset.
+       l1_sensitivity: the sensitivity in L1 norm.
+       l2_sensitivity: the sensitivity in L2 norm.
        budget_weight: Relative weight of the privacy budget allocated to this
          aggregation.
    """
     noise_kind: NoiseKind
-    l0_sensitivity: int
-    linf_sensitivity: float
+    l0_sensitivity: Optional[int] = None
+    linf_sensitivity: Optional[float] = None
+    l1_sensitivity: Optional[float] = None
+    l2_sensitivity: Optional[float] = None
```

Comment:
where do we use these new values in this PR? should be a separate PR with code where we use them?

better to split I think but if we keep then let's at least update PR description